code/modules/mining/mint.dm

```diff
@@ -21,43 +19,43 @@
 	var/coinsToProduce = 10
 
 
-/obj/machinery/mineral/mint/New()
-	..()
-	spawn( 5 )
-		for (var/dir in cardinal)
-			src.input = locate(/obj/machinery/mineral/input, get_step(src, dir))
-			if(src.input) break
-		for (var/dir in cardinal)
-			src.output = locate(/obj/machinery/mineral/output, get_step(src, dir))
-			if(src.output) break
-		START_PROCESSING(SSobj, src)
+/obj/machinery/mineral/mint/Initialize()
+	. = ..()
+	for (var/dir in cardinal)
```

Comment:
I feel like with how often this style of for loop is used it might be convenient to write a helper proc that checks adjacent turfs for a given type.
Or even just a function that compiles the contents of adjacent turfs into a single list it can locate() in. Just and idle thought